File: 03.SortingAlgo/BubbleSort/BubbleSort.cpp
#include <iostream>
using namespace std;

void bubbleSort(int arr[], int n) {
    for(int i = 0; i < n; i++) {
        bool isSwapped = false;
        for(int j = 0; j < n - i - 1; j++) {
            if(arr[j] > arr[j+1]) {
                // Adjacent element swapping
                int temp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = temp;
                isSwapped = true;
            }
        }
        // If no swapping occurred then array is sorted
        if(!isSwapped) return; // Return from function as array is sorted
    }
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int n = sizeof(arr) / sizeof(arr[0]);
    bubbleSort(arr, n);
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
    return 0;
}

File: 03.SortingAlgo/BubbleSort/BubbleSort.js
function bubbleSort(arr) {
    const n = arr.length;
    for (let i = 0; i < n; i++) {
        let isSwapped = false;
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                // Adjacent element swapping
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                isSwapped = true;
            }
        }
        // If no swapping occurred then array is sorted
        if (!isSwapped) return; // Return from function as array is sorted
    }
}

// Example usage:
const arr = [1, 2, 3, 4, 5];
bubbleSort(arr);
console.log(arr); // Output: [1, 2, 3, 4, 5]
